### PR TITLE
feat(ui): load sign-in experience settings from server

### PR DESCRIPTION
**Pull request by simeng-li**
## Summary

Load sign-in experience settings from the server ahead
SSR takes some effort and needs to figure out a better solution.  Add the client-side request ahead. 

- load theme settings from the sign-in experience return value
- rename channel param to method to better match the experience settings naming convention
- keep the AppContent component clean move the context to App top level

## Testing

test locally 
image

@logto-io/eng

### CODE REVIEW

**Review by wangsijie — APPROVED** [review]

File: packages/ui/src/pages/SecondarySignIn/index.tsx [hunk 1]
```diff
@@ -9,31 +9,31 @@ import UsernameSignin from '@/containers/UsernameSignin';
...
-    if (channel !== 'email' && channel !== 'sms' && channel !== 'username') {
+    if (method !== 'email' && method !== 'sms' && method !== 'username') {
       navigate('/404', { replace: true });
     }
-  }, [channel, navigate]);
+  }, [method, navigate]);
 
   const signInForm = useMemo(() => {
-    if (channel === 'sms') {
+    if (method === 'sms') {
```
Comment (IceHe) [inline packages/ui/src/pages/SecondarySignIn/index.tsx:27]: nit: `switch` statement better?

File: packages/ui/src/pages/Register/index.tsx [hunk 2]
```diff
@@ -9,31 +9,31 @@ import { PhonePasswordless, EmailPasswordless } from '@/containers/Passwordless'
...
-    if (channel !== 'email' && channel !== 'sms' && channel !== 'username') {
+    if (method !== 'email' && method !== 'sms' && method !== 'username') {
       navigate('/404', { replace: true });
     }
-  }, [channel, navigate]);
+  }, [method, navigate]);
 
   const registerForm = useMemo(() => {
-    if (channel === 'sms') {
+    if (method === 'sms') {
```
Comment (IceHe) [inline packages/ui/src/pages/Register/index.tsx:27]: nit: switch statement better?

File: packages/ui/src/pages/Register/index.tsx [hunk 3]
```diff
@@ -9,31 +9,31 @@ import { PhonePasswordless, EmailPasswordless } from '@/containers/Passwordless'
...
 
 const Register = () => {
   const { t } = useTranslation(undefined, { keyPrefix: 'main_flow' });
   const navigate = useNavigate();
-  const { channel = 'username' } = useParams<Parameters>();
+  const { method = 'username' } = useParams<Parameters>();
 
   useEffect(() => {
-    if (channel !== 'email' && channel !== 'sms' && channel !== 'username') {
+    if (method !== 'email' && method !== 'sms' && method !== 'username') {
```
Comment (IceHe) [inline packages/ui/src/pages/Register/index.tsx:21]: nit: `!['email', 'sms', 'username'].includes(method)` better?

File: packages/ui/src/apis/utils.ts [hunk 4]
```diff
@@ -15,32 +15,32 @@ import {
 
 export type PasscodeChannel = 'sms' | 'email';
 
-export const getSendPasscodeApi = (type: UserFlow, channel: PasscodeChannel) => {
-  if (type === 'sign-in' && channel === 'email') {
+export const getSendPasscodeApi = (type: UserFlow, method: PasscodeChannel) => {
```
Comment (IceHe) [inline packages/ui/src/apis/utils.ts:18]: seems that always return `sendSignInEmailPasscode` when `method` is `email` and return `sendRegisterSMSPasscode` when `method` is `sms`.

could we simplify it? (maybe do it in another PR)

File: packages/ui/src/apis/utils.ts [hunk 5]
```diff
@@ -15,32 +15,32 @@ import {
...
+  if (type === 'register' && method === 'email') {
     return sendRegisterEmailPasscode;
   }
 
   return sendRegisterSMSPasscode;
 };
 
-export const getVerifyPasscodeApi = (type: UserFlow, channel: PasscodeChannel) => {
-  if (type === 'sign-in' && channel === 'email') {
+export const getVerifyPasscodeApi = (type: UserFlow, method: PasscodeChannel) => {
```
Comment (IceHe) [inline packages/ui/src/apis/utils.ts:34]: ditto

File: packages/ui/src/apis/utils.ts [hunk 6]
```diff
@@ -15,32 +15,32 @@ import {
...
     return sendSignInSMSPasscode;
   }
 
-  if (type === 'register' && channel === 'email') {
+  if (type === 'register' && method === 'email') {
     return sendRegisterEmailPasscode;
   }
 
   return sendRegisterSMSPasscode;
 };
```
Comment (IceHe) [inline packages/ui/src/apis/utils.ts:32]: seems that always return `sendSignInEmailPasscode` when `method` is `email` and return `sendRegisterSMSPasscode` when `method` is `sms`.

could we simplify it? (maybe do it in another PR)

File: packages/ui/src/apis/utils.ts (hunk 5 above)
Comment (IceHe) [inline packages/ui/src/apis/utils.ts:34]: ditto

**Review by IceHe — CHANGES_REQUESTED** [review]

File: packages/ui/src/pages/Register/index.tsx (hunk 2 above)
Comment (simeng-li) [reply packages/ui/src/pages/Register/index.tsx:27]: I think simple logic better avoid using switch

File: packages/ui/src/pages/Register/index.tsx (hunk 3 above)
Comment (simeng-li) [reply packages/ui/src/pages/Register/index.tsx:21]: @IceHe  updated PLTA

File: packages/ui/src/apis/utils.ts (hunk 6 above)
Comment (simeng-li) [reply packages/ui/src/apis/utils.ts]: sorry not following, what do you mean by `always return sendSignInEmailPasscode when method is email` ? There is another API call sendRegisterXXXXPasscode

**Review by IceHe — APPROVED** [review]
lgtm
